File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
  def isPalindrome(self, x: int) -> bool:
    if x < 0:
      return False
    j = x
    n = 0
    while j:
      n += j % 10
      n *= 10
      j //= 10
    return n / 10 == x

File: 0009-palindrome-number/0009-palindrome-number.go
import (
  "strconv"
)

func isPalindrome(x int) bool {
  if x < 0 { return false }
  s := strconv.Itoa(x)
  n := len(s)
  for i := range n / 2 {
    if s[i] != s[n - i - 1] {
      return false
    }
  }
  return true
}